doctests/search_vss.py

```diff
@@ -0,0 +1,340 @@
+#           'id': 'bikes:009',
+#           'payload': None,
+#           'brand': 'Peaknetic',
+#           'model': 'Secto',
+#           'price': '430'
+#       }
+# ]
+# STEP_END
+# REMOVE_START
+assert all(
+    item in [x.__dict__["id"] for x in res] for item in ["bikes:008", "bikes:009"]
+)
+# REMOVE_END
+
+# STEP_START simple_query_3
+query = Query("@brand:Peaknetic @price:[0 1000]").return_fields(
+    "id", "brand", "model", "price"
+)
+res = client.ft("idx:bikes_vss").search(query).docs
+# print(res)
+# >>> [
+#       Document {
+#           'id': 'bikes:009',
+#           'payload': None,
+#           'brand': 'Peaknetic',
+#           'model': 'Secto',
+#           'price': '430'
+#       }
+# ]
+# STEP_END
+# REMOVE_START
+assert all(item in [x.__dict__["id"] for x in res] for item in ["bikes:009"])
+# REMOVE_END
+
+# STEP_START def_bulk_queries
+queries = [
+    "Bike for small kids",
+    "Best Mountain bikes for kids",
+    "Cheap Mountain bike for kids",
+    "Female specific mountain bike",
+    "Road bike for beginners",
+    "Commuter bike for people over 60",
+    "Comfortable commuter bike",
+    "Good bike for college students",
+    "Mountain bike for beginners",
+    "Vintage bike",
+    "Comfortable city bike",
+]
+# STEP_END
+
+# STEP_START enc_bulk_queries
+encoded_queries = embedder.encode(queries)
+len(encoded_queries)
+# >>> 11
+# STEP_END
+# REMOVE_START
+assert len(encoded_queries) == 11
+# REMOVE_END
+
+
+# STEP_START define_bulk_query
+def create_query_table(query, queries, encoded_queries, extra_params=None):
+    """
+    Creates a query table.
+    """
+    results_list = []
+    for i, encoded_query in enumerate(encoded_queries):
+        result_docs = (
+            client.ft("idx:bikes_vss")
+            .search(
+                query,
+                {"query_vector": np.array(encoded_query, dtype=np.float32).tobytes()}
+                | (extra_params if extra_params else {}),
+            )
+            .docs
+        )
+        for doc in result_docs:
+            vector_score = round(1 - float(doc.vector_score), 2)
+            results_list.append(
+                {
+                    "query": queries[i],
+                    "score": vector_score,
+                    "id": doc.id,
+                    "brand": doc.brand,
+                    "model": doc.model,
+                    "description": doc.description,
+                }
+            )
+
+    # Optional: convert the table to Markdown using Pandas
+    queries_table = pd.DataFrame(results_list)
+    queries_table.sort_values(
+        by=["query", "score"], ascending=[True, False], inplace=True
+    )
+    queries_table["query"] = queries_table.groupby("query")["query"].transform(
+        lambda x: [x.iloc[0]] + [""] * (len(x) - 1)
+    )
+    queries_table["description"] = queries_table["description"].apply(
+        lambda x: (x[:497] + "...") if len(x) > 500 else x
+    )
+    queries_table.to_markdown(index=False)
```

Comment:
Given that this is "optional" and nothing is actually even _done_ with the markdown return value, this should probably be removed so `pandas` and `tabulate` aren't required to run the tests.

Reply:
@akx Again, I think it's trying to show the user a real world example. The code sample doesn't use the Markdown explicitly but the output Markdown table is actually shown in the [tutorial page](https://redis.io/docs/latest/develop/get-started/vector-database/#2-perform-a-k-nearest-neighbors-knn-query). I think the doc text on the page could probably use some improvement but that's a separate PR on the docs repo.

Reply:
`to_markdown()` returns the table as a string, so this would just do a lot of work to dataframe-ify things, group them, transform them... and then throw the result away. I really don't think it's a great idea to spend CI CPU cycles (and contribute to our planet burning in a minor way) for that.

I assume these are somehow extracted from something like a notebook session, where the value of the last expression might actually be shown?

Reply:
OK, I've added a return value to the function and then the main code prints that out when the function is called. The doc page shows the Markdown table as rendered, but it looks like it only shows the first query currently (this is an issue with the doc page, not the code sample - I'll fix that separately).

Reply:
I'm sorry, I'm not following. Why bother with that either? Is there something asserting on the output?

If there is, if the idea of these being here (and e.g. being run as part of CI?) is to test that "typical" Redis-utilizing code works, then it would break if `pandas` or `tabulate` decide to change their (human-oriented) output.

If there isn't, I don't see a point in doing the work to human-format data that no one will care about.

(On that note, running `sentence-transformers` in CI also sounds like a bad idea, since it will have to download the transformers models, then run embedding computation on CPU, ...)

Reply:
I think you might be misunderstanding what this code is for :-) The code samples in this folder are included in the docs (eg, https://redis.io/docs/latest/develop/get-started/vector-database/). Their purpose is to explain to the user how to do stuff with Redis and to supply bits of code for copy/paste. The reason for putting them in the client repo as tests is to check that the sample code works against the current version of the client, not to act as real world example tests for the client.

Reply:
@akx , these tests were living in a separate branch until now, this is a first step to put them as close as possible to the code they are documenting. Right now there is no CI in this repo for them, I guess we can add something in other PRs. The purpose so far is educational, and we want to decrease maintenance effort. I'll merge this as it is.